.github/workflows/pull-request-test-harness.yml

```diff
@@ -0,0 +1,105 @@
+name: Pull Request CI - Test harness
+on:
+  # Triggers the workflow on push or pull request events but only for the "main" or "develop" branch
+  pull_request:
+    branches:
+      - main
+    types:
+      - opened
+      - reopened
+      - ready_for_review
+      - synchronize
+    paths:
+      - 'test-harness/**'
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# Required for sam-validate step
+permissions:
+  id-token: write
+  contents: read
+
+env: # Only adding the variables in that are required for
+  AWS_REGION: eu-west-2
+
+jobs:
+  run-code-check:
+    name: Code Checks
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+        working-directory: test-harness/src/
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v3
+        with:
+          submodules: true
+
+      - name: Checks if committing secrets to repo
+        uses: pre-commit/action@v3.0.0
+        with:
+          extra_args: "detect-secrets --all-files"
+
+      - name: Setup nodeJS v18
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+
+      - name: Set offline mirror path
+        id: offline-mirror-path
+        run: echo "mirror-dir=${GITHUB_WORKSPACE}/npm-packages-offline-cache" >> $GITHUB_OUTPUT
+
+      - name: Cache npm offline-mirror
+        uses: actions/cache@v3
+        with:
+          path: ${{ steps.offline-mirror-path.outputs.mirror-dir }}
+          key: offline-mirror-oauth
+
+      - name: Install dependencies
+        run: npm install
+
+      - name: Run lint
+        run: npm run lint
+
+      - name: Run Tests
+        run: npm run test:unit
+
+  sam-validate:
+    name: SAM Validate
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+        working-directory: test-harness
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v3
+        with:
+          submodules: true
+
+      - name: Setup SAM CLI
+        uses: aws-actions/setup-sam@v2
+        with:
+          use-installer: true
+
+      - name: Assume temporary AWS role
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          role-to-assume: ${{ secrets.GH_DEV_TEST_HARNESS_VALIDATE_ROLE_ARN }}
```

Comment:
GH_DEV_TEST_HARNESS_VALIDATE_ROLE_ARN is missing in Github Secrets

Reply:
Raised ticket